lib/swagger/blocks.rb

```diff
@@ -298,45 +462,105 @@ def api(&block)
       end
     end
 
-    # http://goo.gl/PvwUXj#522-api-object
+    # v1.2: http://goo.gl/PvwUXj#522-api-object
     class ApiNode < Node
       def operation(&block)
         self.data[:operations] ||= []
-        self.data[:operations] << Swagger::Blocks::OperationNode.call(&block)
+        self.data[:operations] << Swagger::Blocks::OperationNode.call(version: version, &block)
       end
     end
 
+    # v2.0: https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#path-item-object
+    class PathNode < Node
+      # TODO support ^x- Vendor Extensions
+
+      def operation(op, &block)
+        op = op.to_sym
+        # TODO proper exception class
+        raise "Invalid operation" unless [:get, :put, :post, :delete,
```

Comment:
Could you move this list into a constant and raise an ArgumentError with a bit more detail here? 

Example line from another place:
```ruby
raise ArgumentError.new("#{name} not in #{GRANT_TYPES}") if !GRANT_TYPES.include?(name)
```